vscode/src/completions/context/retrievers/recent-user-actions/recent-view-port.ts

```diff
@@ -42,35 +52,33 @@ export class RecentViewPortRetriever implements vscode.Disposable, ContextRetrie
             window.onDidChangeTextEditorVisibleRanges(
                 debounce(this.onDidChangeTextEditorVisibleRanges.bind(this), 300)
             ),
-            window.onDidChangeActiveTextEditor(this.onDidChangeActiveTextEditor.bind(this))
+            window.onDidChangeNotebookEditorVisibleRanges(
+                debounce(this.onDidChangeNotebookEditorVisibleRanges.bind(this), 300)
+            ),
+            window.onDidChangeActiveTextEditor(this.onDidChangeActiveTextEditor.bind(this)),
+            window.onDidChangeActiveNotebookEditor(this.onDidChangeActiveNotebookEditor.bind(this))
         )
     }
 
     public async retrieve({ document }: ContextRetrieverOptions): Promise<AutocompleteContextSnippet[]> {
         const sortedViewPorts = this.getValidViewPorts(document)
 
         const snippetPromises = sortedViewPorts.map(async viewPort => {
-            const document = await vscode.workspace.openTextDocument(viewPort.uri)
-            const content = document.getText(viewPort.visibleRange)
-
             const snippet: AutocompleteContextSnippet = {
                 uri: viewPort.uri,
-                content,
-                startLine: viewPort.visibleRange.start.line,
-                endLine: viewPort.visibleRange.end.line,
+                content: viewPort.content,
                 identifier: this.identifier,
                 metadata: {
                     timeSinceActionMs: Date.now() - viewPort.lastAccessTimestamp,
                 },
             }
             return snippet
         })
-        const viewPortSnippets = await Promise.all(snippetPromises)
-        return viewPortSnippets
+        return await Promise.all(snippetPromises)
```

Comment:
`await` is redundant here. we can give control of the promise to the consumer right away

```suggestion
        return Promise.all(snippetPromises)
```

Reply:
done